firmware/raspi/boardinterface.py

```diff
@@ -228,16 +232,22 @@ def send_message_to_arduino(self, board_move):
         msg = f"~{board_move.op_code}{source_str}{dest_str}{board_move.move_count % 10}"
 
         print(f"Sending message \"{msg}\" to arduino")
-
-        # TODO: Implement sending message to arduino
+        # TODO: Comment out ser.write(msg) when testing game loop
+        #ser.write(msg)
+        
+        
 
         # TODO: This is for game loop dev, remove once we read from arduino
         self.set_status_from_arduino(ArduinoStatus.EXECUTING_MOVE, board_move.move_count, None)
 
     def get_status_from_arduino(self):
         '''Read status from Arduino over UART connection.
         '''
-        # TODO: update this function to actually read from arduino
+        #New variable created, new_input, to store 4 bytes for UART Messages
+        #If the start byte is a ~ and the Arduino Status is valid, process the arduino status based on the new input
+        new_input = ser.read(4)
+        if new_input[0] == '~' and ArduinoStatus.is_valid_code(new_input[1]):
+            self.arduino_status = ArduinoStatus(new_input[1], new_input[3], new_input[2])
```

Comment:
For now, let's please leave the `ser.read(4)` and the if statement commented out. We can uncomment it when we're ready to run this on the Pi.